collect_app/src/main/java/org/odk/collect/android/preferences/FormManagementPreferences.java

```diff
@@ -77,40 +84,59 @@ public void onCreate(Bundle savedInstanceState) {
         setupFormUpdateMode();
     }
 
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+        super.onSharedPreferenceChanged(sharedPreferences, key);
+
+        if (key.equals(KEY_FORM_UPDATE_MODE) || key.equals(KEY_PERIODIC_FORM_UPDATES_CHECK)) {
```

Comment:
Yeah I think we could use this for a lot more. Analytics is a great shout as it would let us capture more from the imports. I'd like to take a proper pass over some of these preference screens again as I was avoiding too many changes due to the AndroidX stuff being in flight. I'll add a note to my project backlog.

As to the split feeling a little weird: I agree. It might be that local changes like that are just better handles using the individual preference listeners.